prdoc/pr_4281.prdoc

```diff
@@ -0,0 +1,16 @@
+title: "Add support for versioned notification for HRMP pallet"
+
+doc:
+  - audience: Runtime Dev
+    description: |
+      The configuration of the HRMP pallet has been expanded to include the `VersionWrapper` type,
+      which controls the encoding of XCM notifications related to the opening/closing of HRMP channels.
+      If your runtime does not concern itself with the XCM version used for notifications,
+      you can set it as `type VersionWrapper = ()` to always use the latest XCM.
+      If your runtime does care about the XCM version when sending to child parachains,
```

Comment:
So if your runtime is a parachain you just put `()`, right?
What does it mean to concern oneself about versions? Is it allowing older versions of XCM?

Reply:
> So if your runtime is a parachain you just put `()`, right? 

The HRMP pallet is dedicated to relay chain runtimes, I don't think that any parachain runtime uses this HRMP pallet.


> What does it mean to concern oneself about versions? Is it allowing older versions of XCM?

The problem could arise from a higher version on the relay chain than on the receiving parachain. For example, imagine that we update Polkadot with `xcm::v5`, but Polimec still uses `xcm::v4`. Without this fix, they could lose messages because they would not be able to decode `xcm::v5`.

Reply:
Perhaps it could be possible to replace most of the functionality related to the WrapVersion with just the `SendXcm` trait. In this scenario, the HRMP pallet would use `type XcmSender = ChildParachainRouter`, potentially simplifying things. However, this approach would introduce another layer of complexity, particularly regarding handling transport fees. Nevertheless, I think these HRMP notifications are treated as system notifications, which is why they are sent directly by `dmp::Pallet::<T>::queue_downward_message`.